utils/ipprefix/ipprefix.go

```diff
@@ -0,0 +1,166 @@
+package ipprefix
+
+import (
+	"encoding/binary"
+	"net"
+)
+
+//IPcache is an interface which provides functionality to store ip's and do longest prefix match
+type IPcache interface {
+	Put(net.IP, int, interface{})
+	Get(net.IP, int) (interface{}, bool)
+	RunVal(func(val interface{}) interface{})
+	RunIP(net.IP, func(val interface{}) bool)
+}
+
+const (
+	ipv4MaskSize = 32 + 1
+	ipv6MaskSize = 128 + 1
+)
+
+type ipcache struct {
+	ipv4 []map[uint32]interface{}
+	ipv6 []map[[16]byte]interface{}
+}
+
+//NewIPCache creates an object which is implementing the interface IPcache
+func NewIPCache() IPcache {
+	return &ipcache{
+		ipv4: make([]map[uint32]interface{}, ipv4MaskSize),
+		ipv6: make([]map[[16]byte]interface{}, ipv6MaskSize),
+	}
+}
+
+// Put takes an argument an ip address, mask and value. It is used as a cache for quick lookup
+func (cache *ipcache) Put(ip net.IP, mask int, val interface{}) {
+	if ip.To4() != nil {
+		ip = ip.To4()
+
+		if cache.ipv4[mask] == nil {
+			cache.ipv4[mask] = map[uint32]interface{}{}
+		}
+
+		m := cache.ipv4[mask]
+		// the following expression is ANDing the ip with the mask
+		m[binary.BigEndian.Uint32(ip)&binary.BigEndian.Uint32(net.CIDRMask(mask, 32))] = val
+	} else {
```

Comment:
return and avoid the else. I suspect the linter will complain as well. Same in the below methods.